oracle_gis.c

```diff
@@ -725,9 +726,14 @@ char *
 ewkbHeaderFill(oracleSession *session, ora_geometry *geom, char * dest)
 {
 	const unsigned srid = ewkbSrid(session, geom);
-	const ub1 flags = ((3 == ewkbDimension(session, geom)) ? 0x01 : 0x00 );
+	ub1 flags;
 	ub1 s[3];
 
+	if (ewkbIsMeasured(session, geom))
+		flags = 0x02;  /* two-dimensional geometry with measure */
+	else
+		flags = ((3 == ewkbDimension(session, geom)) ? 0x01 : 0x00 );
+
```

Comment:
I see the problem; I think I have not been clear.
``ewkbDimension`` will return Oracle's idea of dimension (I have added a function comment to that extent), so in the case of a ``POINT M`` this would be 3.
The comment you are referring to describes what ``0x02`` means in PostGIS, so maybe I should improve the comment.

Probing around I noticed that there is no check for four-dimensional Oracle geometries in the code at all. Currently such geometries lead to errors like

    ERROR:  oracle_fdw internal error: number of bytes written 28 is different from size 44

Not nice at all. I will add a check with a later commit.